discovery/consul/consul.go

```diff
@@ -457,6 +463,7 @@ func (d *Discovery) watchService(ctx context.Context, ch chan<- []*targetgroup.G
 		labels: model.LabelSet{
 			serviceLabel:    model.LabelValue(name),
 			datacenterLabel: model.LabelValue(d.clientDatacenter),
+			namespaceLabel:  model.LabelValue(d.clientNamespace),
```

Comment:
Can we use serviceNode.Service.Namespace ? So  it would be retrieved per service even if the user does not specify it.

Reply:
As you can only connect to a singe namespace per discovery, do we need that? I wasn't sure about that part of the implementation so I followed `datacenterLabel`.

Also: This should always be empty for open source Consul (there are no namespaces), for Consul Enterprise it's `default` if no namespace is specified by the user.

Reply:
Empty labels are fine, and it seems that then we would get natively a default namespace if we do not specify it in consul entreprise, so it seems like a win to me?

Reply:
friendly ping :)

Reply:
I hear you … 😉 I still want to check if the current implementation always returns the desired result.

Reply:
@roidelapluie Ok, switched to `serviceNode.Service.Namespace` instead. Sorry, it took me a bit to understand what you were getting at. 😅